### DISCUSSION TITLE
config file

### DISCUSSION THREAD

**Discussion by viveksharmatws** [tailwindlabs/tailwindcss#12280] [Help]
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./src/**/*.{js,ts,jsx,tsx,html}",
    "./public/*.{js,ts,jsx,tsx,html}",
  ],
  theme: {
    fontFamily: {
      primary: ["Outfit", "sans-serif"],
    },
    extend: {
      colors: {
        background: {
          DEFAULT: "#F6F7FB",
          primary: "#B2BECD",
          secondary: "#F6F7FB",
        },
        primary: {
          DEFAULT: "#364862",
        },
        secondary: {
          DEFAULT: "#29B6F6",
          hover: "#5DC8F8",
        },
      },
    
      
    },
    fontSize: {
      "7xl":["3.5rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "6xl":["2.5rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "5xl":["2rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "4xl":["1.75rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "3xl":["1.5rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "2xl":["1.375rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      "c-21":[ "1.3125rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }], // custom 21px
      "c-19":[ "1.1875rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }], // custom
      xl: ["1.25rem",{
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      l:[ "1.125rem", {
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
      m:[ "1rem", {
        lineHeight: "",
        letterSpacing :"",
        fontWeight : ""
      }],
    },
  },
};
trying to make the custome patterns of   fontweights, tracking,leading with fontsize names    is there is anything wrong witht the code and do I need to put fonsizes inside the extend

**Comment by wongjn** [comment]
> is anything wrong witht the code

You'd want to add CSS values for each of the `lineHeight`, `letterSpacing` and `fontWeight` object properties if you want to have your own values for them per each `fontSize` value. At the moment, they are empty strings.

>  do I need to put fonsizes inside the extend

You do not necessarily need to place your `fontSize` configuration inside `theme.extend`. Placing it `theme` means the default Tailwind configuration for `fontSize` is **cleared**, and these default Tailwind values won't be available. Contrastingly, having your `fontSize` configuration inside `theme.extend` would still make the default Tailwind configuration for `fontSize` available, but with any matching value names – like `7xl` – *your* values will take precedence.
